src/app/d2-loadout-builder/generated-sets/utils.ts

```diff
@@ -47,34 +41,39 @@ export function getSetsForTier(
 
   // Prioritize list based on number of matched perks
   Object.keys(lockedMap).forEach((bucket) => {
-    // if there are locked items for this bucket
-    if (lockedMap[bucket] && lockedMap[bucket].items.length && lockedMap[bucket].type === 'perk') {
-      // Sort based on what sets have the most matched perks
-      matchedSets.sort((a, b) => {
-        return (
-          sum(b.armor, (item) => {
-            if (!item || !item.sockets) {
-              return 0;
-            }
-            return item.sockets.sockets.filter((slot) =>
-              slot.plugOptions.some((perk) =>
-                lockedMap[bucket].items.find((lockedPerk) => lockedPerk.hash === perk.plugItem.hash)
-              )
-            ).length;
-          }) -
-          sum(a.armor, (item) => {
-            if (!item || !item.sockets) {
-              return 0;
-            }
-            return item.sockets.sockets.filter((slot) =>
-              slot.plugOptions.some((perk) =>
-                lockedMap[bucket].items.find((lockedPerk) => lockedPerk.hash === perk.plugItem.hash)
-              )
-            ).length;
-          })
-        );
-      });
+    // if there are locked perks for this bucket
+    if (lockedMap[bucket] === undefined) {
+      return;
+    }
+    const lockedPerks = lockedMap[bucket].filter((lockedItem) => lockedItem.type === 'perk');
+    if (!lockedPerks.length) {
+      return;
     }
+    // Sort based on what sets have the most matched perks
+    matchedSets.sort((a, b) => {
```

Comment:
I'd suggest using `_.sortBy`, which besides being less code, will be more efficient since it computes the sort value with one run through the array rather than each time a comparison needs to happen.